### ISSUE TITLE
`process.exited?` throws an error when process is detached

### ISSUE THREAD

**Issue by jawshooah** [jarib/childprocess#84]
I'd like to be able to check whether a detached `ChildProcess` has exited, but invoking `process.exited?` throws an error:

```ruby
>> process = ChildProcess.build('touch', 'some-file')
=> #<ChildProcess::Unix::ForkExecProcess:0x007fa4a49643a8 @args=["touch", "some-file"], @started=false, @exit_code=nil, @io=nil, @cwd=nil, @detach=false, @duplex=false, @leader=false, @environment={}>
>> process.detach = true
=> true
>> process.start
=> #<ChildProcess::Unix::ForkExecProcess:0x007fa4a49643a8 @args=["touch", "some-file"], @started=true, @exit_code=nil, @io=nil, @cwd=nil, @detach=true, @duplex=false, @leader=false, @environment={}, @pid=19628>
>> process.exited?
Errno::ECHILD: No child processes
	from /Users/joshuahagins/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/childprocess-0.5.5/lib/childprocess/unix/process.rb:32:in `waitpid2'
	from /Users/joshuahagins/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/childprocess-0.5.5/lib/childprocess/unix/process.rb:32:in `exited?'
	from (irb):14
	from /Users/joshuahagins/.rbenv/versions/2.2.1/bin/irb:11:in `<main>'
```

**Comment by jarib** [comment]
Released this fix as 0.5.6.

**Comment by jawshooah** [comment]
Excellent, thanks!
